aplus/celery.py

```diff
@@ -29,3 +30,5 @@ def enroll():
     )
     for i in courses:
         i.enroll_from_sis()
+        if settings.SIS_ENROLL_DELAY:
+            sleep(settings.SIS_ENROLL_DELAY)
```

Comment:
> This sleeps even if the course `i` does not use Sisu integration. At least this could be changed so that the task can finish faster. The total duration would increase with every course instance added to the database even if only few courses use the Sisu integration.

The 'courses' query should include only those courses that are ongoing and that have sis_enroll enabled.

> Sleeping prevents the Celery worker from running other tasks while waiting. However, I didn't find any alternative, thus in order to throttle the Sisu API requests, `sleep` is needed. What about the other Celery tasks? Do we have other workers for running tasks when one worker is completely occupied by this sleeping scheduled task?

I understood it possible to use multiple concurrent threads for worker ('concurrency' parameter), which should probably be set to more than 1.

> Did you consider the (total) time limits for a Celery task? If there is a time limit, it could kill this task before reaching the end. https://docs.celeryproject.org/en/stable/reference/celery.app.task.html#celery.app.task.Task.soft_time_limit

As far as I know, we haven't specified any time limit, and it should default to no limit.